src/email_requests/views.py

```diff
@@ -65,8 +66,17 @@ class ListOwnRequestsView(generics.ListAPIView):
     serializer_class = RequestSerializer
 
     def get_queryset(self):
-        mentor = get_object_or_404(Mentor, profile__user=self.request.user)
-        return Request.objects.filter(mentor=mentor).order_by('date_created').reverse()
+        profile = get_object_or_404(Profile, user=self.request.user)
+
+        try:
+            mentor = Mentor.objects.get(profile=profile)
+        except Mentor.DoesNotExist:
+            mentor = None
+
+        if (mentor!=None):
+            return Request.objects.filter(Q(mentor=mentor) | Q(mentee=profile)).order_by('date_created').reverse()
+        else:
+            return Request.objects.filter(mentee=profile).order_by('date_created').reverse()
```

Comment:
you can combine these two statements as such

query = Q(mentee=profile)

if (mentor is not None) :
query |= Q(mentor=mentor) 

Return Request...